### ISSUE TITLE
Develop CLI Style guide

### ISSUE THREAD

**Issue by klizhentas** [gravitational/teleport#12811]
Review Heroku CLI style guide and develop our own:

https://devcenter.heroku.com/articles/cli-style-guide

**Comment by zmb3** [comment]
I would love to see some of the advice from https://seirdy.one/posts/2022/06/10/cli-best-practices/ incorporated.

**Comment by zmb3** [comment]
Example of inconsistent use of `--force` flag: https://github.com/gravitational/teleport/issues/1872

**Comment by Tener** [comment]
I know this could be a large undertaking, but if we are looking into improving our CLI then I think considering the switch from kingpin to cobra is a worthwhile thing (https://github.com/spf13/cobra, https://cobra.dev/). I know things were different a few years back, but right now cobra sees *much* more activity: https://go.libhunt.com/compare-kingpin-vs-cobra.

**Comment by zmb3** [comment]
Unlikely to happen as the `tsh aliases` work is being done on our fork of kingpin.

**Comment by Tener** [comment]
> Unlikely to happen as the `tsh aliases` work is being done on our fork of kingpin.

I haven't tried it, but my gut feeling is that my `tsh aliases` PR wouldn't be substantially different with cobra: https://github.com/gravitational/teleport/pull/13305. 

That being said, cobra is not a drop-in replacement, so some issues will likely appear in one place or another.

**Comment by GavinFrazar** [comment]
@smallinsky do you mind if I work on this? I think our CLI really needs refinement. It's the major entry-point to setting up and using Teleport, yet we have a lot of inconsistencies and issues with it. The first step in cleaning it up should be establishing our own rules around CLI code style and UX as this feature-request suggests. Then we can work to move our CLI into compliance with those rules, and then setting up lints and tests to keep us in compliance with the rules so we don't drift away from them as other work is done.

I've come across many more issues while working on https://github.com/gravitational/teleport/issues/19560 that make me think this should be prioritized.

**Comment by zmb3** [comment]
This should also include the use of colors, see #36929 for example.

**Comment by GavinFrazar** [comment]
> This should also include the use of colors, see #36929 for example.

good point, also mentioned by https://seirdy.one/posts/2022/06/10/cli-best-practices/

> Excessive animation and color can harm users with attention and/or vestibular disorders, and some on the autism spectrum. Many tools offer a --color[=WHEN] flag where WHEN is always, never, or auto. Expecting users to learn all the color configurations for all their tools is unrealistic; tools should [respect the NO_COLOR environment variable.](https://no-color.org/)
